src/main/java/com/aws/greengrass/deployment/DeploymentQueue.java

```diff
@@ -112,6 +113,20 @@ public synchronized boolean offer(Deployment offeredDeployment) {
         return true;
     }
 
+    private boolean checkReplacementCriteria(Deployment enqueued, Deployment offered) {
+        // if the enqueued deployment is in non-DEFAULT stage, then do not replace
+        if (!Deployment.DeploymentStage.DEFAULT.equals(enqueued.getDeploymentStage())) {
+            return false;
+        }
+        // if the offered deployment is a new Shadow deployment, then it supersedes existing shadow deployment
+        // if the offered deployment is a cancelled deployment, then it should replace the existing deployment
+        if (Deployment.DeploymentType.SHADOW.equals(offered.getDeploymentType()) || offered.isCancelled()) {
+            return true;
+        }
+        // if the offered deployment is in non-DEFAULT stage, then replace it
```

Comment:
replace "it" with "the enqueued deployment"

Reply:
aren't we replacing the enqueued one with the offered one?